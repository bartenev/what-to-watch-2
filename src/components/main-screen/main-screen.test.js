import React from "react";
import renderer from "react-test-renderer";
import MainScreen from "./main-screen";

const films = [
  {
    description: `Mauris interdum dolor a convallis pharetra. Fusce porttitor dictum mi, id tincidunt ligula lobortis eu. Integer commodo enim eget ullamcorper faucibus. Lorem ipsum dolor sit amet, consectetur adipiscing elit. Integer commodo enim eget ullamcorper faucibus. Lorem ipsum dolor sit amet, consectetur adipiscing elit. `,
    director: `Christopher Nolan`,
    genre: `Action`,
    rating: {
      score: 5,
      count: 783,
      level: `Excellent`,
    },
    released: 1956,
    reviews: [
      {
        date: `23/1/2021`,
        name: `Ivan`,
        rating: 8,
        text: `Mauris interdum dolor a convallis pharetra. Fusce porttitor dictum mi, id tincidunt ligula lobortis eu. Vivamus elementum nisl eu elit cursus, id luctus justo posuere. Vivamus elementum nisl eu elit cursus, id luctus justo posuere. Nunc sapien turpis, eleifend in odio at, ullamcorper efficitur lorem. `,
      },
      {
        date: `13/3/2021`,
        name: `Genka`,
        rating: 0,
        text: `Nunc sapien turpis, eleifend in odio at, ullamcorper efficitur lorem. Quisque fringilla lacinia pulvinar. Donec vel semper nunc, ac posuere lorem. Nulla sed tortor nunc.. `,
      },
      {
        date: `17/6/2021`,
        name: `Cheba`,
        rating: 5,
        text: `Mauris interdum dolor a convallis pharetra. Fusce porttitor dictum mi, id tincidunt ligula lobortis eu. Integer commodo enim eget ullamcorper faucibus. Lorem ipsum dolor sit amet, consectetur adipiscing elit. `,
      },
      {
        date: `28/6/2021`,
        name: `Cheba`,
        rating: 5,
        text: `Nunc sapien turpis, eleifend in odio at, ullamcorper efficitur lorem. Praesent in nisi condimentum, aliquet nisl at, fermentum neque. Nunc sapien turpis, eleifend in odio at, ullamcorper efficitur lorem. `,
      }
    ],
    runTime: 240,
    src: {
      poster: `img/pulp-fiction.jpg`,
      preview: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
      video: ``,
    },
    starring: [
      `Quentin Tarantino`,
      `David Fincher`,
    ],
    title: `Pulp Fiction`
  },
  {
    description: `Nunc sapien turpis, eleifend in odio at, ullamcorper efficitur lorem. Vivamus ex urna, tincidunt eget feugiat a, luctus ut arcu. Mauris interdum dolor a convallis pharetra. Fusce porttitor dictum mi, id tincidunt ligula lobortis eu. `,
    director: `Christopher Nolan`,
    genre: `Mystery`,
    rating: {
      score: 0,
      count: 6354,
      level: `Excellent`,
    },
    released: 1966,
    reviews: [
      {
        date: `21/4/2021`,
        name: `Vasya`,
        rating: 4,
        text: `Mauris interdum dolor a convallis pharetra. Fusce porttitor dictum mi, id tincidunt ligula lobortis eu. Nunc sapien turpis, eleifend in odio at, ullamcorper efficitur lorem. `,
      },
    ],
    runTime: 220,
    src: {
      poster: `img/macbeth.jpg`,
      preview: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
      video: ``,
    },
    starring: [
      `Robert Zemeckis`,
      `Christopher Nolan`,
      `Clint Eastwood`,
    ],
    title: `Macbeth`,
  },
];

it(`Render MainScreen`, () => {
  const tree = renderer.create((
    <MainScreen
      films={films}
      onWelcomeButtonClick={() => {}}
    />
  ), {
    createNodeMock: () => {
      return {};
    }
  }).toJSON();

  expect(tree).toMatchSnapshot();
});

