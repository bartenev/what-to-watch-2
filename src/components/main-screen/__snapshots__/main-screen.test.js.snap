// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render MainScreen 1`] = `
<div
  className="page-content"
>
  <section
    className="catalog"
  >
    <h2
      className="catalog__title visually-hidden"
    >
      Catalog
    </h2>
    <ul
      className="catalog__genres-list"
    >
      <li
        className="catalog__genres-item catalog__genres-item--active"
      >
        <a
          className="catalog__genres-link"
          href="#"
          onClick={[Function]}
        >
          All genres
        </a>
      </li>
      <li
        className="catalog__genres-item "
      >
        <a
          className="catalog__genres-link"
          href="#"
          onClick={[Function]}
        >
          Action
        </a>
      </li>
      <li
        className="catalog__genres-item "
      >
        <a
          className="catalog__genres-link"
          href="#"
          onClick={[Function]}
        >
          Mystery
        </a>
      </li>
    </ul>
    <div
      className="catalog__movies-list"
    >
      <article
        className="small-movie-card catalog__movies-card"
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        <div
          className="small-movie-card__image"
        >
          <video
            className="player__video"
            muted={true}
          />
        </div>
        <h3
          className="small-movie-card__title"
        >
          <a
            className="small-movie-card__link"
            href="movie-page.html"
            onClick={[Function]}
          >
            Pulp Fiction
          </a>
        </h3>
      </article>
      <article
        className="small-movie-card catalog__movies-card"
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        <div
          className="small-movie-card__image"
        >
          <video
            className="player__video"
            muted={true}
          />
        </div>
        <h3
          className="small-movie-card__title"
        >
          <a
            className="small-movie-card__link"
            href="movie-page.html"
            onClick={[Function]}
          >
            Macbeth
          </a>
        </h3>
      </article>
    </div>
    
  </section>
  <footer
    className="page-footer"
  >
    <div
      className="logo"
    >
      <a
        className="logo__link logo__link--light"
      >
        <span
          className="logo__letter logo__letter--1"
        >
          W
        </span>
        <span
          className="logo__letter logo__letter--2"
        >
          T
        </span>
        <span
          className="logo__letter logo__letter--3"
        >
          W
        </span>
      </a>
    </div>
    <div
      className="copyright"
    >
      <p>
        Â© 2019 What to watch Ltd.
      </p>
    </div>
  </footer>
</div>
`;
